kernel/cost.h

```diff
@@ -43,13 +57,12 @@ struct CellCosts
 			{ ID($_AOI4_),   7 },
 			{ ID($_OAI4_),   7 },
 			{ ID($_MUX_),    4 },
-			{ ID($_NMUX_),   4 }
+			{ ID($_NMUX_),   4 },
+			{ ID($_DFF_P_),  1 },
+			{ ID($_DFF_N_),  1 },
```

Comment:
We should probably add all of the DFF and latch types here, but I don't know what a reasonable cost estimate for them would be (1 seems off).

Reply:
We can look at the sky130hd and asap7 areas for those and the other cells

Reply:
I'd say that's out of scope. Want me to make an issue? I added these only because I noticed stat.cc was patching these on its side so I consolidated it into here

Reply:
What is using these costs? Is there any information about where they come from and what they are supposed to model?

Reply:
`stat.cc` uses CMOS transistor estimates of these (16). Nothing uses default gate count estimates of these. I was using this logic: by being primitives to be techmapped to, they have a default gate count of 1

Reply:
> by being primitives to be techmapped to, they have a default gate count of 1

I am not sure I follow. Wouldn't that make the cost 1 for all the primitives in the list?

Reply:
I think it may be appropriate to use a model suitable for pre-synthesis estimates.
So something like:
1. buffers and inverters have no cost as they aren't full 'functions'
2. simple 2-input gates (AND, NAND, OR, NOR, MUX...) have a cost of 1
3. complex 2-input gates (XOR, XNOR) have a cost of 2
4. larger gates use the equivalent of what they would use, were you to implement them as 2-input gates
This is the same approach used in [Zimmermann's thesis](https://iis-people.ee.ethz.ch/~zimmi/publications/adder_arch.pdf), you need to scale it to whatever base you use (times four if you count transistors).  
Sequential elements (all of them) should likely be substantially more expensive because it is (usually) harder to optimize them away and depending on your technology, you have additional costs related to this cell (special placement considerations, it may use an additional metal layer internally, it may need decap cells nearby, more flops -> more clock tree etc).

In my opinion it may even make sense to split combination and sequential cost especially considering the application in `keep_hierarchy` since you may want to keep modules which aren't big but have a lot of state (eg register files).